examples/eks_cluster_custom_iam/iam.tf

```diff
@@ -0,0 +1,225 @@
+
+# Create instance profile for CAST AI created nodes
+resource "aws_iam_instance_profile" "castai_instance_profile" {
+  name = local.instance_profile_role_name
+  role = aws_iam_role.castai_instance_profile_role.name
+}
+
+# Create instance profile role
+resource "aws_iam_role" "castai_instance_profile_role" {
+  name               = local.instance_profile_role_name
+  assume_role_policy = jsonencode({
+    Version : "2012-10-17"
+    Statement : [
+      {
+        "Sid"     = ""
+        Effect    = "Allow"
+        Principal = {
+          Service = "ec2.amazonaws.com"
+        }
+        Action = ["sts:AssumeRole"]
+      }
+    ]
+  })
+}
+
+# Attach policies to instance profile
+resource "aws_iam_role_policy_attachment" "instance_profile_policy" {
+  for_each = toset([
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+  ])
+
+  role       = aws_iam_instance_profile.castai_instance_profile.role
+  policy_arn = each.value
+}
+
+################################################################################
+# Assume role
+################################################################################
+
+# Create role that will be assumed by CAST AI user
+resource "aws_iam_role" "assume_role" {
+  name               = local.iam_role_name
+  assume_role_policy = jsonencode({
+    Version   = "2012-10-17"
+    Statement = [
+      {
+        Action    = "sts:AssumeRole"
+        Effect    = "Allow"
+        Principal = {
+          AWS = data.castai_eks_user_arn.castai_user_arn.arn
+        }
+      },
+    ]
+  })
+}
+
+# Attach readonly policies to role
+resource "aws_iam_role_policy_attachment" "assume_role_readonly_policy_attachment" {
+  for_each = toset([
+    "arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess",
+    "arn:aws:iam::aws:policy/IAMReadOnlyAccess",
+  ])
+  role       = aws_iam_role.assume_role.name
+  policy_arn = each.value
+}
+
+# Attach inline policy to role
+resource "aws_iam_role_policy" "inline_role_policy" {
+  name   = local.iam_inline_policy_name
+  role   = aws_iam_role.assume_role.name
+  policy = jsonencode({
+    Version   = "2012-10-17"
+    Statement = [
+      # Needed to be able launch instance with instance profile
+      {
+        Sid       = "PassRoleEC2",
+        Action    = "iam:PassRole",
+        Effect    = "Allow",
+        Resource  = "arn:aws:iam::${local.account_id}:role/${aws_iam_role.castai_instance_profile_role.name}",
+        Condition = {
+          StringEquals = {
+            "iam:PassedToService" = "ec2.amazonaws.com"
+          }
+        }
+      },
+      # Needed to tag non CAST nodes
+      {
+        Sid    = "NonResourcePermissions",
+        Effect = "Allow",
+        Action = [
+          "ec2:CreateTags",
+          # Necessary only if public SSH key for CAST nodes will be used
+          "ec2:ImportKeyPair",
+        ],
+        Resource = "*"
```

Comment:
Could we have a narrower scope here?

Reply:
yes and no, importKeyPair is optional anyway and createTags is not destructive action